### PR TITLE
Fix volume mount without volume

### PR DESCRIPTION
**Pull request by RafalKorepta**
When internal communication to Admin API is configured to have mutual TLS configuration the volumes and volume mounts need to have additional mtls-client certificate included. That certificate could be generated in volume mounts if internal Admin API has disabled mutual TLS, but any other internal listener has it enabled.

### CODE REVIEW

File: charts/redpanda/chart_template_test.go [hunk 1]
```diff
@@ -118,6 +123,36 @@ func TestTemplate(t *testing.T) {
 	}
 }
 
+func AssertVolumeMountsVerification(t *testing.T, manifests []byte) {
+	objs, err := kube.DecodeYAML(manifests, redpanda.Scheme)
+	require.NoError(t, err)
+
+	for _, obj := range objs {
+		sts, ok := obj.(*v1.StatefulSet)
```
Comment (chrisseto) [inline charts/redpanda/chart_template_test.go:131]: I think we'll be doing ourselves a lot of favors if we make this work for more than just `StatefulSets`. If you think that's too much work for now can this be renamed to showcase that it's statefulset specific? Something like `ASSERT-STATEFULSET-VOLUME-MOUNTS-VERIFICATION`.

We could recursively crawl the members of a struct looking for PodTemplates using `reflect` 😅

File: charts/redpanda/chart_template_test.go [hunk 2]
```diff
@@ -18,6 +18,7 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"golang.org/x/tools/txtar"
+	v1 "k8s.io/api/apps/v1"
```
Comment (chrisseto) [inline charts/redpanda/chart_template_test.go:21]: nit: For consistency let's import this as `appsv1`

**Review by chrisseto — APPROVED** [review]

File: charts/redpanda/chart_template_test.go (hunk 2 above)
Comment (RafalKorepta) [reply charts/redpanda/chart_template_test.go:21]: ok

File: charts/redpanda/chart_template_test.go (hunk 1 above)
Comment (RafalKorepta) [reply charts/redpanda/chart_template_test.go:131]: I changed the annotation. I will leave enhancing to find PodTemplates using `reflect` for later :)

**Pull request merged**